UnityProject/Assets/Scripts/Messages/Server/HighlanderTimerMessage.cs

```diff
@@ -0,0 +1,32 @@
+using InGameEvents;
+using Mirror;
+using UI;
+using UnityEngine;
+
+namespace Messages.Server
+{
+	public class HighlanderTimerMessage : ServerMessage<HighlanderTimerMessage.NetMessage>
+	{
+		public struct NetMessage : NetworkMessage
+		{
+			public float Time;
+		}
+
+		public override void Process(NetMessage msg)
+		{
+			HighlanderTimerUI.Instance.Show(msg.Time);
+		}
+
+		public static void Send(ConnectedPlayer player)
+		{
+			var highlanderEvent =  InGameEventsManager.Instance.ListOfAntagonistEventScripts.Find(x => x.GetType() == typeof(EventHighlander))
```

Comment:
what is this?

Reply:
It gets the event so it grabs the time thats on it and send it to the player

Reply:
how often? It also has a GetComponent

Reply:
only once, it doesn't get spammed.

Reply:
hewwwooo?